types/wlr_buffer.c

```diff
@@ -49,7 +49,7 @@ void wlr_buffer_unlock(struct wlr_buffer *buffer) {
 		return;
 	}
 
-	assert(buffer->n_locks > 0);
+	assert(buffer->n_locks > 0 || !buffer->dropped);
```

Comment:
like that:

Can I change 

````
static void buffer_consider_destroy(struct wlr_buffer *buffer) {
	if (!buffer->dropped || buffer->n_locks > 0) {
		return;
	}
````

To

````
static void buffer_consider_destroy(struct wlr_buffer *buffer) {
	if (!buffer->dropped && buffer->n_locks > 0) {
		return;
	}
````

?

Reply:
No, because this will destroy the buffer under one user's nose.

There are two users of `wlr_buffer`:

- The buffer creator has ownership over the buffer. It calls `wlr_buffer_drop` when it's done.
- Multiple consumers can keep references to the buffer via `wlr_buffer_lock`, and call `wlr_buffer_unlock` when they are done.

If the creator is done with the buffer, but a consumer still has a reference to it, we can't destroy the buffer. The other way around is true too.

Reply:
OK, I will update this commit tomorrow